src/cmd/services/m3dbnode/main/main.go

```diff
@@ -53,21 +53,36 @@ func main() {
 		os.Exit(1)
 	}
 
-	dbClientCh := make(chan client.Client, 1)
-	clusterClientCh := make(chan clusterclient.Client, 1)
+	var (
+		dbClientCh        chan client.Client
+		clusterClientCh   chan clusterclient.Client
+		coordinatorDoneCh chan struct{}
+	)
+
+	if cfg.DB != nil {
+		dbClientCh = make(chan client.Client, 1)
+		clusterClientCh = make(chan clusterclient.Client, 1)
+	}
+
 	if cfg.Coordinator != nil {
+		coordinatorDoneCh = make(chan struct{}, 1)
 		go func() {
 			coordinatorserver.Run(coordinatorserver.RunOptions{
 				Config:        *cfg.Coordinator,
 				DBClient:      dbClientCh,
 				ClusterClient: clusterClientCh,
 			})
+			coordinatorDoneCh <- struct{}{}
 		}()
 	}
 
-	dbserver.Run(dbserver.RunOptions{
-		Config:          cfg.DB,
-		ClientCh:        dbClientCh,
-		ClusterClientCh: clusterClientCh,
-	})
+	if cfg.DB != nil {
+		dbserver.Run(dbserver.RunOptions{
+			Config:          *cfg.DB,
+			ClientCh:        dbClientCh,
+			ClusterClientCh: clusterClientCh,
+		})
+	} else if cfg.Coordinator != nil {
+		<-coordinatorDoneCh
+	}
```

Comment:
Should we print a warning if neither are configured?

Reply:
I'm ok for now to just quit at the moment, we can revisit this later - its a little difficult because we don't have a logger at this point and stderr/stdout is gross when at all other times we would use a logger.